crates/revm/revm-inspectors/src/tracing/builder/parity.rs

```diff
@@ -192,10 +192,10 @@ impl ParityTraceBuilder {
         self,
         res: ResultAndState,
         trace_types: &HashSet<TraceType>,
-        db: DB,
+        db: &mut DB,
     ) -> Result<TraceResults, DB::Error>
     where
-        DB: DatabaseRef,
+        DB: Database,
```

Comment:
these should not take `Database`, they keep using DatabaseRef imo

because `Database::basic` behaves differently than `DatabaseRef::basic`

https://github.com/bluealloy/revm/issues/540

Reply:
Reverted back to `CacheDb` inside rpc